Sources/BraintreeCore/RepeatingTimer.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+
+final class RepeatingTimer {
+    
+    /// Amount of time, in seconds.
+    private let timeInterval: Int
+    
+    init(timeInterval: Int) {
+        self.timeInterval = timeInterval
+    }
+    
+    private lazy var timer: DispatchSourceTimer = {
+        let timerSource = DispatchSource.makeTimerSource(queue: DispatchQueue.main)
+        timerSource.schedule(
+            deadline: .now() + .seconds(timeInterval),
+            repeating: .seconds(timeInterval),
+            leeway: .seconds(1)
+        )
+        timerSource.setEventHandler(handler: { [weak self] in
+            self?.eventHandler?()
+        })
+        return timerSource
+    }()
+    
+    var eventHandler: (() -> Void)?
+    
+    private enum State {
+        case suspended
+        case resumed
+    }
+    
+    private var state: State = .suspended
```

Comment:
Curious why we need to manage this state ourselves?

Reply:
GCD timers can be somewhat sensitive. If you try and `resume`/`suspend` an already `resumed`/`suspended` timer, you will get a crash with a reason like:

> BUG IN CLIENT OF LIBDISPATCH: Over-resume of an object

This tells us we have tried to resume an already resumed timer. We just need to balance the calls to suspend and resume, like the [documentation states](https://developer.apple.com/library/content/documentation/General/Conceptual/ConcurrencyProgrammingGuide/GCDWorkQueues/GCDWorkQueues.html#//apple_ref/doc/uid/TP40008091-CH103-SW8):

> As a result, you must balance each call to dispatch_suspend with a matching call to dispatch_resume before event delivery resumes.

Reply:
Gotcha, thanks for the background here. This makes sense

Reply:
Can we include that link & those details in a comment?